File: tech_interview_practice/reverse_string/reverse.py
def stack_way(text):
    stack = []
    for x in text:
        stack.append(x)
    reverse = ""
    while stack:
        reverse += stack.pop()
    return reverse

def recursive_way(text):
    if len(text) <= 1:
        return text
    return recursive_way(text[1:]) + text[0]

def simple_way(text):
    return text[::-1]

original = input('Enter some text: ')
print(stack_way(original))
print(recursive_way(original))
print(simple_way(original))

File: interview_practice/reverse_string/reverse.cpp
#include <iostream>
#include <string>

using namespace std;

string reverse(string text) {
  if (text.size() <= 1) {
    return text;
  }
  return reverse(text.substr(1)) + text[0];
}

int main() {
  string input;
  cout << "Enter some text: ";
  getline(cin, input);
  cout << reverse(input) << endl;
  return 0;
}
